tests/registration_test.py

```diff
@@ -533,6 +533,114 @@ def test_strings_groupby_attach(self):
         self.assertIsInstance(sAttach.segments, ak.pdarray)
         self.assertIsInstance(sAttach.unique_keys, ak.Strings)
 
+    def test_string_property_accessor(self):
```

Comment:
As a general note, I think it would be good to add some comments to indicate what each section is intended to test. It makes coming back to things like this were there are a lot of cases easier to identify what is going on.